lib/policies/rpRegistrationPolicy.ts

```diff
@@ -1,21 +1,32 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
-import { BaseFilter } from "./baseFilter";
 import { HttpOperationResponse } from "../httpOperationResponse";
-import { WebResource } from "../webResource";
 import * as utils from "../util/utils";
+import { WebResource } from "../webResource";
+import { BaseRequestPolicy, RequestPolicyCreator, RequestPolicy, RequestPolicyOptions } from "./requestPolicy";
 
 /* tslint:disable:prefer-const */
 let retryTimeout = 30;
 /* tslint:enable:prefer-const */
 
-export class RPRegistrationFilter extends BaseFilter {
+export function rpRegistrationPolicy(retryTimeout = 30): RequestPolicyCreator {
+  return (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
+    return new RPRegistrationPolicy(nextPolicy, options, retryTimeout);
+  };
+}
+
+export class RPRegistrationPolicy extends BaseRequestPolicy {
 
-  constructor(retryTimeout = 30) {
-    super();
+  constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions, retryTimeout = 30) {
+    super(nextPolicy, options);
     retryTimeout = retryTimeout;
```

Comment:
This isn't part of your work, but this smells funny and should be revisited.

Reply:
Agreed. I'll take care of that right now.

Reply:
Done.